sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/ModelHelper.java

```diff
@@ -74,7 +75,7 @@ public static TableEntity createEntity(Map<String, Object> properties) {
                 new IllegalStateException("'entityCreator' should not be null."));
         }
 
-        return entityCreator.apply(properties);
+        return entityCreator.get().addProperties(properties);
```

Comment:
Can this entityCreateor return a null item?

Reply:
No, thankfully. It calls the TableEntity constructor, and the only way that can fail is if an exception is thrown.